src/ducks/classifications.js

```diff
@@ -121,10 +123,81 @@ const createClassification = () => {
   };
 };
 
+const saveAllQueuedClassifications = (dispatch) => {
+  const queue = JSON.parse(localStorage.getItem(FAILED_CLASSIFICATION_QUEUE_NAME));
+  if (queue && queue.length !== 0) {
+    console.log('Saving queued classifications:', queue.length);
+    queue.forEach((classificationData) => {
+      apiClient.type('classifications').create(classificationData).save()
+        .then((actualClassification) => {
+          console.log('Saved classification', actualClassification.id);
+
+          try {  //Fix: IE11 doesn't know what to do with Split.classificationCreated()
+            Split.classificationCreated(actualClassification);
+          } catch (err) { console.error('Split.classificationCreated() error: ', err); }
+
+          actualClassification.destroy();
+          const indexInQueue = queue.indexOf(classificationData);
+          queue.splice(indexInQueue, 1);
```

Comment:
Here's the tricky thing; `queue.splice()` changes the very array that we're iterating through via `forEach()`.

Imagine...
```
var queue = ['a', 'b', 'c', 'd'];
queue.forEach(function (item, index) {
  console.log('item: ' + item + ', index: ' + index);
  queue.splice(index, 1);
});
```
...what do we expect the output to be?

A.
```
item: a, index: 0
item: b, index: 1
item: c, index: 2
item: d, index: 3
```

or B.
```
item: a, index: 0
item: c, index: 1
```

From what I can see, we _want_ the logic for `saveAllQueuedClassifcations` to iterate through _every_ item in the queue, so we want result A. However, `queue.splice()` is the equivalent of sawing the branch you're sitting on, so what we get is result B.

Basically, the in-place update of the `queue` has a risk of 'skipping' Classifications; what we need is a separate `queueToIterateThrough` and `queueToSaveBackToLocalhost`

Reply:
e.g.
```
NEW_QUEUE = [];
queue.forEach((classificationData) => {
  apiClient.type('classifications').create(classificationData).save()
  .then((actualClassification) => {
    ...
    //If successful, don't do queue.splice()
    ...
  })
  .catch((err) => {
    ...
    //If UNSUCCESSFUL, return it to the queue
    NEW_QUEUE.push(actualClassification);  //Uh, gotta check if this function can actually see the actualClassification variable
  }
  ...
}

localStorage.setItem(FAILED_CLASSIFICATION_QUEUE_NAME, JSON.stringify(NEW_QUEUE));
```

Reply:
This is a really good point. The code here seems to be copied from the PFE coffeescript, but I'm not sure if the PFE code is any good.